src/main/java/com/nextstep/lotto/view/LottoMain.java

```diff
@@ -1,21 +1,65 @@
 package com.nextstep.lotto.view;
 
+import com.nextstep.lotto.domain.Lotto;
 import com.nextstep.lotto.domain.LottoSeller;
 import com.nextstep.lotto.domain.Lottos;
+import com.nextstep.lotto.domain.WinningLotto;
 
+import java.util.ArrayList;
 import java.util.List;
+import java.util.Optional;
+import java.util.function.Supplier;
 
 public class LottoMain {
     public static void main(String[] args) {
         LottoMain.run();
     }
 
     public static void run() {
-        int price = InputView.inputPrice();
-        Lottos lottos = LottoSeller.buy(price);
-        ResultView.printBuy(lottos);
-        List<Integer> winnerNumbers = InputView.inputWinnerNumbers();
-        int bonusNumber = InputView.inputBonusNumber();
-        ResultView.printStatistics(lottos.getStatistics(winnerNumbers, bonusNumber));
+        Lottos lottos = buy();
+        draw(lottos);
+    }
+
+    private static Lottos buy() {
+        int totalCount = inputWithoutException(InputView::inputPriceToCount);
+        int manualCount = inputWithoutException(() -> InputView.inputManualCount(totalCount));
+        int autoCount = totalCount - manualCount;
+
+        List<Lotto> manualLottos = inputManualLottos(manualCount);
+        Lottos lottos = LottoSeller.buy(manualLottos, autoCount);
+        ResultView.printBuy(manualCount, autoCount, lottos);
+        return lottos;
+    }
+
+    private static List<Lotto> inputManualLottos(int manualCount) {
+        List<Lotto> manualLottos = new ArrayList<>();
+        for ( int i = 0 ; i < manualCount ; i ++ ) {
+            Lotto manualLotto = inputWithoutException(InputView::inputManualLotto);
+            manualLottos.add(manualLotto);
+        }
+        return manualLottos;
+    }
+
+    private static void draw(Lottos lottos) {
+        WinningLotto winningLotto = inputWithoutException(InputView::inputWinningLotto);
+        ResultView.printStatistics(lottos.getStatistics(winningLotto));
+    }
+
+    public static <T> T inputWithoutException(Supplier<T> supplier) {
```

Comment:
재귀호출 활용하면 좀더 간략하게 짤 수 있습니다.
그런데 저는 별로 추천하고 싶지 않은 예외처리 방법이네요
try catch를 예외처리를 위한 용도가 아닌 다른 용도로 쓰게 되면 문제 생겼을 때 파악하기 어렵습니다.

```suggestion
    private static <T> T inputWithoutException(Supplier<T> supplier) {
        try {
            return supplier.get();
        } catch ( Exception e ) {
            System.out.println(e.getMessage());
            return inputWithoutException(supplier);
        }
    }
```

Reply:
스펙상 재귀호출을 해도 무한히 exception이 발생 할 가능성이 있는 상태입니다. 코드야 조금 간략해 지겠지만 스택 오버플로우 가능성을 열어두는건 좋은 선택이 아니지 않을까요 ? 🤔 
catch에서 println으로 찍고 먹어버리는 것 때문에 말씀주신 것 같은데, 보편적으로는 맞는 말입니다만 현재 구현은 예외를 'cause는 출력 하되 throw는 하지 않게' 의도한 것이기 때문에 괜찮지 않나 싶습니다. 어떻게 생각하시나요 ?

Reply:
스택 오버플로우 가능성도 있고 try catch 용도에 맞지 않아서 저도 추천하는 방법은 아니라고 생각합니다.
지금처럼 구현 했을 때 아래와 같은 이슈가 있는데 더 나은 방법이 없을지 고민해보는 것도 좋을 것 같습니다

1. 예외 처리를 위해 optional을 null 처리가 아닌 다른 용도로 사용하고 있다
2. optional을 다른 의도로 사용하다보니 코드 파악이 어렵다
3. try catch를 용도에 맞지 않게 사용하고 있다. 진짜 버그가 발생했을때 버그인지 의도한 코드인지 파악이 어렵다